presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainOrcPredicate.java

```diff
@@ -432,19 +457,50 @@ private static Filter doubleRangeToFilter(Range range, boolean nullAllowed)
     {
         Marker low = range.getLow();
         Marker high = range.getHigh();
-        double lowerDouble = low.isLowerUnbounded() ? Double.MIN_VALUE
-                : ((Double) low.getValue()).doubleValue();
-        double upperDouble = high.isUpperUnbounded() ? Double.MAX_VALUE
-                : ((Double) high.getValue()).doubleValue();
-        return new Filters.DoubleRange(lowerDouble,
+        double lowerDouble = low.isLowerUnbounded() ? Double.MIN_VALUE : ((Double) low.getValue()).doubleValue();
```

Comment:
I don't understand why here we will cast to boxed type Double, then call `doubleValue`, where above code like that has been reverted.  Why not just let it be `Double` and `Long`, and let autoboxing work its magic in the Filters constructors below.

Reply:
@tdcmeehan How about `double upperDouble = high.isUpperUnbounded() ? Double.MAX_VALUE : (double) high.getValue();`?